src/python/grpcio_tests/tests_aio/unit/client_streaming_interceptor_test.py

```diff
@@ -0,0 +1,314 @@
+                return call
+
+            @property
+            def response_iterator(self):
+                return self._response_iterator
+
+        interceptor = Interceptor()
+
+        channel = aio.insecure_channel(self._server_target, interceptors=[interceptor])
+        stub = test_pb2_grpc.TestServiceStub(channel)
+
+        request = messages_pb2.StreamingOutputCallRequest()
+        for _ in range(_NUM_STREAM_RESPONSES):
+            request.response_parameters.append(
+                messages_pb2.ResponseParameters(size=_RESPONSE_PAYLOAD_SIZE))
+
+        call = stub.StreamingOutputCall(request)
+
+        response_cnt = 0
+        async for response in call:
+            response_cnt += 1
+            self.assertIs(type(response),
+                          messages_pb2.StreamingOutputCallResponse)
+            self.assertEqual(_RESPONSE_PAYLOAD_SIZE, len(response.payload.body))
+
+        self.assertTrue(response_cnt, _NUM_STREAM_RESPONSES)
+        self.assertTrue(interceptor.response_iterator.response_cnt, _NUM_STREAM_RESPONSES)
+        self.assertEqual(await call.code(), grpc.StatusCode.OK)
+
+        await channel.close()
+
+    async def test_intercepts_response_iterator_using_read(self):
+
+        class Interceptor(aio.UnaryStreamClientInterceptor):
+
+            def __init__(self):
+                self._response_iterator = None
+
+            async def intercept_unary_stream(self, continuation,
+                                             client_call_details, request, response_iterator):
+                self._response_iterator = ResponseIterator(response_iterator)
+                call = await continuation(client_call_details, request, self._response_iterator)
+                return call
+
+            @property
+            def response_iterator(self):
+                return self._response_iterator
+
+        interceptor = Interceptor()
+
+        channel = aio.insecure_channel(self._server_target, interceptors=[interceptor])
+        stub = test_pb2_grpc.TestServiceStub(channel)
+
+        request = messages_pb2.StreamingOutputCallRequest()
+        for _ in range(_NUM_STREAM_RESPONSES):
+            request.response_parameters.append(
+                messages_pb2.ResponseParameters(size=_RESPONSE_PAYLOAD_SIZE))
+
+        call = stub.StreamingOutputCall(request)
+
+        response_cnt = 0
+        for response in range(_NUM_STREAM_RESPONSES):
+            response = await call.read()
+            response_cnt += 1
+            self.assertIs(type(response),
+                          messages_pb2.StreamingOutputCallResponse)
+            self.assertEqual(_RESPONSE_PAYLOAD_SIZE, len(response.payload.body))
+
+        self.assertTrue(response_cnt, _NUM_STREAM_RESPONSES)
+        self.assertTrue(interceptor.response_iterator.response_cnt, _NUM_STREAM_RESPONSES)
+        self.assertEqual(await call.code(), grpc.StatusCode.OK)
+
+        await channel.close()
```

Comment:
Ideas for test cases:
1. Exception raised in interceptor;
2. Multiple interceptors.